### PR TITLE
fix: change order of eventtypes and routingforms

### LINKED ISSUE #17614: [CAL-4568] add new routers and event-types to the top, not the bottom
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4568](https://linear.app/calcom/issue/CAL-4568/add-new-routers-and-event-types-to-the-top-not-the-bottom)</sub>

### PR DESCRIPTION
**Pull request by vijayraghav-io**
## What does this PR do?

- Fixes #17614 
- Fixes CAL-4568

/claim #17614 

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] - N/A -I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by Praashh — DISMISSED** [review]
LGTM!!

**Comment by vijayraghav-io** [comment]
@PeerRich , updated for eventTypes, routingForms, workflows. 
Bookings already have sort as per the tab.
https://github.com/calcom/cal.com/blob/1aac70f643835960963907d3c48d01ccec9b653c/packages/lib/bookings/getAllUserBookings.ts#L110-L114

As of now, I couldn't find anything more to be included.
Please let know if any more entities to be included.

**Comment by vijayraghav-io** [comment]
@Praashh , all tests are passing

**Comment by Praashh** [comment]
> @Praashh , all tests are passing. you can approve again

let me give it a final touch.

**Review by Praashh — DISMISSED** [review]
LGTM!! thank you so much @vijayraghav-io

**Review by keithwillcode — CHANGES_REQUESTED** [review]
@vijayraghav-io @Praashh Just double-checking we are sure there aren't pieces of code that depend on the sort order for some reason?

(Marking requested changes just to block merging first)

**Comment by vijayraghav-io** [comment]
> @vijayraghav-io @Praashh Just double-checking we are sure there aren't pieces of code that depend on the sort order for some reason?

Hi @keithwillcode , double checked again that - the sole purpose of these updated (sort order) functions is to ultimately render the lists <li in UI . Verified that these functions are not called anywhere else.
Also there is no dependency on sort order in the UI for these entities.

**Comment by keithwillcode** [comment]
> > @vijayraghav-io @Praashh Just double-checking we are sure there aren't pieces of code that depend on the sort order for some reason?
> 
> Hi @keithwillcode , double checked again that - the sole purpose of these updated (sort order) functions is to ultimately render the lists <li in UI through trpc handlers/endpoints. Verified that these functions have no references or are not called anywhere else. Also there is no dependency on sort order in the UI for these entities.

Great. Thanks, @vijayraghav-io

**Review by keithwillcode — DISMISSED** [review]

**Comment by vijayraghav-io** [comment]
Thank you! @keithwillcode 🙏

**Comment by vijayraghav-io** [comment]
Just guessing, If retaining the same look and feel w.r.t order of default event-types for new/future onboarding customers is a concern, then the order can be updated here (or the array can be reversed) -
https://github.com/calcom/cal.com/blob/f9afca2939a72922bfb39c1b00d242c6a37c4136/apps/web/components/getting-started/steps-views/UserProfile.tsx#L81-L98

**Comment by vijayraghav-io** [comment]
updated due to changes after https://github.com/calcom/cal.com/pull/17689 was merged

**Comment by vijayraghav-io** [comment]
Added a new test - to ensure new eventType added will be first in the list

File: apps/web/playwright/embed-code-generator.e2e.ts
Comment (Praashh) [inline apps/web/playwright/embed-code-generator.e2e.ts:1]: I think it's out of the scope of this PR.

**Review by Praashh — CHANGES_REQUESTED** [review]
We also need a test case for the routing form.

File: apps/web/playwright/embed-code-generator.e2e.ts
Comment (vijayraghav-io) [reply apps/web/playwright/embed-code-generator.e2e.ts:1]: the changes here is required.
As the eventTypes order in the list is changed, hence the first expected eventType needs update in these tests, else these tests will fail.

**Review by Praashh — APPROVED** [review]
Great Work @vijayraghav-io sir, LGTM!!!

**Pull request merged**